.github/workflows/preview.yaml

```diff
@@ -30,18 +30,29 @@ jobs:
         uses: actions/github-script@v7
         with:
           script: |
-            const response = await github.rest.search.issuesAndPullRequests({
-              q: 'repo:${{ github.repository }} is:pr sha:${{ github.event.workflow_run.head_sha }}',
-              per_page: 1,
-            })
-            const items = response.data.items
-            if (items.length < 1) {
-              console.error('No PRs found')
-              return
+            async function findPR() {
+              for (let retries = 3; retries > 0; retries--) {
+                try {
+                  const response = await github.rest.search.issuesAndPullRequests({
+                    q: 'repo:${{ github.repository }} is:pr sha:${{ github.event.workflow_run.head_sha }}',
+                    per_page: 1,
+                  });
+                  const items = response.data.items;
+                  if (items.length < 1) {
+                    throw new Error('No PRs found');
+                  }
+                  const pullRequestNumber = items[0].number
+                  console.info("Pull request number is", pullRequestNumber);
+                  return pullRequestNumber
+                } catch (error) {
+                  console.error(`Retrying... (${3 - retries + 1}/3)`);
+                  if (retries === 1) throw error;
+                  await new Promise(res => setTimeout(res, 10000));
+                }
+              }
             }
-            const pullRequestNumber = items[0].number
-            console.info("Pull request number is", pullRequestNumber)
-            return pullRequestNumber
+            return findPR();
```

Comment:
Is there a reason to wrap this in a function?

Reply:
Maybe wrapping it in a function is not needed here, but the intention was to make it easier to manage, to debug, and easier to simulate different scenarios to make sure the retry logic works as expected. But maybe it’s overkill? Let me know if you prefer it a certain way

Reply:
I think it's better to just set `retries` on the action, it will handle the `Retry-After` header for us
https://docs.github.com/en/rest/using-the-rest-api/best-practices-for-using-the-rest-api#handle-rate-limit-errors-appropriately

Reply:
Gotcha. Thanks for the fix!